pyscraper/regmem/wales/process.py

```diff
@@ -103,7 +112,36 @@ def get_ms_details(ms_id: int, *, lang: Literal["en", "cy"] = "en") -> RegmemPer
     headers = {"User-Agent": "TWFY interests scraper"}
     content = requests.get(url, headers=headers).content
 
-    items = pd.read_html(content)
+    max_attempts = 3
+    current_attempt = 0
+
+    while current_attempt < max_attempts:
+        try:
+            items = pd.read_html(content)
+            break  # If successful, exit the loop
+        except ValueError as e:
+            if "No tables found" in str(e):
+                # If this is the first attempt and it failed with "No tables found",
+                # wait 10 seconds and retry
+                print(
+                    f"No tables found for MS ID {ms_id} in {lang} language. Retrying in 10 seconds..."
+                )
+                time.sleep(10)
+                # Get fresh content for the retry
+                content = requests.get(url, headers=headers).content
+                current_attempt += 1
+            if current_attempt == max_attempts:
+                # If this is the last attempt, raise the error
+                print(
+                    f"Failed to find tables for MS ID {ms_id} in {lang} language after {max_attempts} attempts."
+                )
+                raise ScraperError(
+                    f"Failed to find tables for MS ID {ms_id} in {lang} language after {max_attempts} attempts."
+                )
+
+            else:
```

Comment:
I think this else should go with the first if and not the second otherwise this will immediately re-raise the error on the first occurrence.

Reply:
I'll indent it - then I think this works.